components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/CertificateValidationUtil.java

```diff
@@ -819,13 +821,27 @@ public static List<String> getAIALocations(X509Certificate cert) throws Certific
     private static OCSPReq generateOCSPRequest(X509Certificate issuerCert, BigInteger serialNumber)
             throws CertificateValidationException {
 
-        // Add provider BC
-        Security.addProvider(new BouncyCastleProvider());
         try {
+            String providerName = ServerConfiguration.getInstance().getFirstProperty(ServerConstants.JCE_PROVIDER);
```

Comment:
Can't we utilize the getJCEProvider method for this also?

Reply:
Fixed